tests/utils/test_plotting.py

```diff
@@ -29,6 +29,31 @@ def test_paired_histogram_specification_bins(
         assert np.all(bins == expected_bins)
 
 
+@pytest.mark.parametrize(
+    "bad_data", [([[]]), ([[], []]),],
+)
+def test_paired_histogram_specification_bins_raises(bad_data: List):
+    """`_extract_paired_histogram_specification` raises a ValueError on empty data"""
+    for density in [False, True]:
+        with pytest.raises(ValueError):
+            rasa.utils.plotting._extract_paired_histogram_specification(
+                bad_data,
+                num_bins=2,
+                density=density,
+                x_pad_fraction=0,
+                y_pad_fraction=0,
+            )
+
+
+@pytest.mark.parametrize(
+    "bad_data", [([[]]), ([[], []]),],
+)
+def test_plot_paired_histogram_doesnt_raise(bad_data: List):
+    """Empty data shouldn't raise an error."""
+    for density in [False, True]:
+        rasa.utils.plotting.plot_paired_histogram(bad_data, title="", density=density)
```

Comment:
could test for the user warning https://docs.pytest.org/en/6.2.x/warnings.html#warns